scripts/asm-installer/install_asm

```diff
@@ -1899,13 +1913,34 @@ configure_package() {
   if [[ -n "${_CI_ASM_IMAGE_LOCATION}" ]]; then
     run kpt cfg set asm anthos.servicemesh.hub "${_CI_ASM_IMAGE_LOCATION}"
   fi
+  if [[ -n "${_CI_ASM_IMAGE_TAG}" ]]; then
+    run kpt cfg set asm anthos.servicemesh.tag "${_CI_ASM_IMAGE_TAG}"
+  fi
+
   if [[ "${USE_HUB_WIP}" -eq 1 ]]; then
     run kpt cfg set asm gcloud.project.environProjectID "${ENVIRON_PROJECT_ID}"
     run kpt cfg set asm anthos.servicemesh.hubMembershipID "${HUB_MEMBERSHIP_ID}"
   fi
   if [[ -n "${CA_NAME}" && "${CA}" = "gcp_cas" ]]; then
     run kpt cfg set asm anthos.servicemesh.external_ca.ca_name "${CA_NAME}"
   fi
+
+  if [[ "${CA}" == "mesh_ca" && -n "${_CI_TRUSTED_GCP_PROJECTS}" ]]; then
+    # Gather the trust domain aliases from projects.
+    TRUST_DOMAIN_ALIASES="${PROJECT_ID}.svc.id.goog"
+    while read -d ',' -r trusted_gcp_project; do
+      TRUST_DOMAIN_ALIASES="${TRUST_DOMAIN_ALIASES} ${trusted_gcp_project}.svc.id.goog"
+    done <<EOF
+${_CI_TRUSTED_GCP_PROJECTS}
+EOF
+    # kpt treats words in quotes as a single param, while kpt need ${TRUST_DOMAIN_ALIASES} to be splitting params for a list. If we remove quotes, the lint will complain.
+    # eval will translate the quoted TRUST_DOMAIN_ALIASES into params to workaround both.
+    run eval kpt cfg set asm anthos.servicemesh.trustDomainAliases "${TRUST_DOMAIN_ALIASES}"
+  fi
+
+  if [[ -n "${_CI_ENVIRON_PROJECT_NUMBER}" ]]; then
+    run kpt cfg set asm gcloud.project.environProjectNumber "${_CI_ENVIRON_PROJECT_NUMBER}"
+  fi
```

Comment:
Can you move this up where the other environProjectNumber setter is so that it's more obvious what gets overwritten?